### DISCUSSION TITLE
Git branching strategy and app versioning/deployment!!!

### DISCUSSION THREAD

**Discussion by erkand-imeri** [vercel/turborepo#1494] [Help] [answered]
Hello everyone,

I know this question is not strictly turborepo-related but it's still a thing which we need to handle when dealing with the monorepo approach.

So, my question is to know more about how do you guys handle the git branching strategy, how do you deal in monorepo. Do you have aside the main/master branch per each app separately a branch reflecting the app name and acting as develop branch per each app separately?

And what about semver versioning or deployment tags, do you use atlassian changesets, and how do you handle it?

Best regards, Erkand.

**Comment by anthonyshew** [comment, marked as answer]
Generally, `main` should track with the deployment history of your applications. The goal is to always have HEAD be the source of truth using a trunk-based deployment strategy.

Versioning then becomes a question of "do you care about it?" Some teams leave their Internal Package versions at `0.0.0` since it doesn't impact their understanding of their monorepo (everything is always current). Some teams will pay attention to their versions (especially when publishing packages).

**Reply by anthonyshew** [reply]
However, there are plenty of ways of handling these things! What works best for your teams is always the right answer.
